tests/run_tests.py

```diff
@@ -140,6 +149,37 @@ def _print_skip_message(test, skip_tag):
         print("%s (%s) ... skipped '%s'" % (test_name, test_module, reason))
 
 
+def _should_skip(distro=None, version=None, arch=None, reason=None):
+    # all these can be lists or a single value, so covert everything to list
+    if distro is not None and type(distro) is not list:
+        distro = [distro]
+    if version is not None and type(version) is not list:
+        version = [version]
+    if arch is not None and type(arch) is not list:
+        arch = [arch]
+
+    if (distro is None or DISTRO in distro) and (version is None or VERSION in version) and \
+       (arch is None or ARCH in arch):
```

Comment:
Please add a comment about where the global variables are defined and why (to not do it for every test, I guess).

Reply:
Comments added.